src/app/donation-start/donation-start-form/donation-tipping-slider/donation-tipping-slider.component.ts

```diff
@@ -0,0 +1,214 @@
+
+
+  containerClass: string;
+  derivedPercentage: number;
+  tipAmount: number;
+  currencyFormatted: string;
+
+  documentMouseupUnlistener: () => void;
+  documentMousemoveUnlistener: () => void;
+  documentTouchmoveUnlistener: () => void;
+
+  isMoving = false;
+  max: number;
+  pageX: any;
+  mousePos: number;
+  position: number;
+  disableDefaults: boolean = false;
+
+  constructor(
+    public renderer: Renderer2
+  ) {
+    this.documentMouseupUnlistener = renderer.listen('document', 'mouseup', () => {
+      this.isMoving = false;
+    });
+
+    this.documentMousemoveUnlistener = renderer.listen('document', 'mousemove', (event: MouseEvent | TouchEvent) => {
+      this.move(event);
+    });
+
+    this.documentTouchmoveUnlistener = renderer.listen('document', 'touchmove', (event: MouseEvent | TouchEvent) => {
+      this.move(event);
+    });
+  };
+
+
+  ngOnInit() {
+    this.containerClass = 'container space-below-' + this.spaceBelow;
+    this.calcAndSetPercentage();
+    this.adjustDonationPercentageAndValue();
+    this.updateHandlePosition(undefined)
+  }
+
+  ngAfterContentInit() {
+    if (this.bar) {
+      this.bar.nativeElement.addEventListener('mousedown',  (e: MouseEvent | TouchEvent) => {
+        this.isMoving = true;
+        this.move(e);
+      });
+
+      this.bar.nativeElement.addEventListener('touchstart', (e: MouseEvent | TouchEvent) => {
+        this.isMoving = true;
+        this.move(e);
+      });
+
+      this.bar.nativeElement.addEventListener('touchend', () => {
+        this.isMoving = false;
+      });
+    }
+  };
+
+  // detect changes in the donationAmount input
+  ngOnChanges(changed: any) {
+    this.max = this.bar.nativeElement.offsetWidth - this.handle.nativeElement.offsetWidth;
+    if (changed.donationAmount.currentValue != changed.donationAmount.previousValue) {
+      this.calcAndSetPercentage();
+      this.calcAndSetTipAmount();
+      this.adjustDonationPercentageAndValue();
+      this.updateHandlePosition(undefined);
+      this.onHandleMoved(this.derivedPercentage, this.tipAmount);
+    }
+  }
+
+  ngOnDestroy() {
+    this.documentMousemoveUnlistener();
+    this.documentMouseupUnlistener();
+    this.documentTouchmoveUnlistener();
+  }
+
+  calcAndSetPercentage() {
+  // the calculation results from mouse click
+   if (this.isMoving) {
+    this.derivedPercentage = Math.round((this.position / this.max) * this.percentageEnd >= 1 ? (this.position / this.max) * this.percentageEnd : 1);
+   }
+  // the calculation results from input changes
+   else if (!this.disableDefaults){
+      if (!this.donationAmount) {
+        this.derivedPercentage = 0;
+      }
+      if (this.donationAmount >= 1000) {
+        this.derivedPercentage = 7.5;
+      } else if (this.donationAmount >= 300) {
+        this.derivedPercentage = 10;
+      } else if (this.donationAmount >= 100) {
+        this.derivedPercentage = 12.5;
+      } else {
+        this.derivedPercentage = 15;
+      }
+    }
+  }
```

Comment:
logic above definitely looks worth testing. Did we do the thing where we don't set the default percentage after someone choose moves the slider?